ember-toucan-core/src/components/form/file-input-field.hbs

```diff
@@ -0,0 +1,112 @@
+<div
+  class="flex flex-col {{if @isDisabled 'text-disabled'}}"
+  data-root-field={{if @rootTestSelector @rootTestSelector}}
+>
+  <Form::Field as |field|>
+    <field.Label for={{field.id}} data-label>
+      {{@label}}
+      {{#if @hint}}
+        <field.Hint data-hint>{{@hint}}</field.Hint>
+      {{/if}}
+
+      <field.Control class="flex">
+        <div
+          class="bg-overlay-1 shadow-focusable-outline focus-within:shadow-focus-outline relative inline-flex h-8 w-full flex-wrap items-center rounded-sm"
+        >
+          <input
+            aria-describedby="{{if @error field.errorId}}"
+            {{! template lint thinks this input is a button, and so doesn't not support aria-invalid }}
+            {{! template-lint-disable no-unsupported-role-attributes }}
+            aria-invalid={{if @error "true"}}
+            class="text-titles-and-attributes focus:outline-none absolute m-1 h-6 w-0 cursor-pointer bg-transparent p-1
+              {{if @isDisabled 'text-disabled' 'text-titles-and-attributes'}}
+              {{if @error 'shadow-error-outline' 'shadow-focusable-outline'}}"
+            disabled={{@isDisabled}}
+            id={{field.id}}
+            type="file"
+            accept={{this.accept}}
+            multiple={{this.multiple}}
+            {{! template-lint-disable no-bare-strings no-inline-styles  }}
+            style="clip: rect(1px, 1px, 1px, 1px);"
+            ...attributes
+            {{on "change" this.handleChange}}
+          />
+          <div
+            class="text-titles-and-attributes inline-flex w-full cursor-pointer items-center gap-x-2 pl-2"
+          >
+            {{! TODO: https://github.com/CrowdStrike/ember-toucan-core/issues/49 replace with official icon later }}
+            <svg
+              xmlns="http://www.w3.org/2000/svg"
+              width="24"
+              height="24"
+              stroke="currentColor"
+              viewBox="0 0 24 24"
+            ><path
+                d="M3.4 6.2h17.2m-15.3 0V19a2.063 2.063 0 00.538 1.413A1.836 1.836 0 007.2 21h9.6a1.861 1.861 0 001.325-.587A2.039 2.039 0 0018.7 19V6.2M10 15.6v-4m4 0v4M9.4 6.2V3.9a.98.98 0 01.212-.637A.746.746 0 0110.2 3h3.6a.743.743 0 01.587.263.975.975 0 01.213.637v2.3"
+                fill="none"
+                stroke-linecap="round"
+                stroke-linejoin="round"
+                stroke-width="2"
+              ></path></svg>
+            {{#if (has-block "triggerText")}}
+              {{yield to="triggerText"}}
```

Comment:
I don't have a strong opinion on string-based args vs. named blocks, but it seems this is a bit inconsistent now, as we we have `@label` (and no `{{yield to="label"}}`), but no `@triggerText`. 

Is there a reason for going this route here? Like do we expect trigger text to not just be a string, but some more complex HTML maybe?

Also, it doesn't seem to be mentioned in the docs! (only in the demo)

Reply:
<img width="697" alt="Screen Shot 2023-03-09 at 12 35 42 PM" src="https://user-images.githubusercontent.com/771170/224109784-bfe58b52-af00-4ef0-a42c-058f7e1399a8.png">

These are all good points! Perhaps I should revisit what I wrote in the docs.

For context, this is a bit easier to explain with pictures...the "Select files" is particular to file inputs, which have both a `label`, then a `button label` then a `description for the button` (like `Select files`)

So looks like
1. I should possibly rename `triggerText`, but not sure what
2. update docs to explain difference between `@label` and what ends up being `triggerText`

Reply:
Thanks for that helpful explanation!

But just to be clear here, my point was not that we don't need both "triggerText" and label. My point was: is there a reason that triggerText is a named block (which could accept more than a plain string, like arbitrary HTML), but label is an argument (only accepting strings)?

At first it seems a bit inconsistent to me, but maybe there is a good reason for this?